### PR TITLE
Fix wpdb::get_results() return type

### PR DESCRIPTION
**Pull request by herndlm**
Based on https://github.com/php-stubs/wordpress-stubs/pull/145

Closes https://github.com/php-stubs/wordpress-stubs/issues/143 

I'll do the list in another follow-up, the changes are bigger than I thought and blow the PR up unnecessarily I think

### CODE REVIEW

**Comment by herndlm** [comment]
fyi @IanDelMar, also, thx @IanDelMar  and @szepeviktor  for working on this yesterday!

File: tests/data/wpdb.php [hunk 1]
```diff
@@ -14,8 +14,19 @@
...
+ *                       Any of ARRAY_A | ARRAY_N | OBJECT | OBJECT_K constants.
+ *                       ARRAY_A | ARRAY_N | OBJECT: return an array of rows indexed from 0 by SQL result row number.
+ *
+ *                       ARRAY_A: Each row is an associative array (column => value, ...),
+ *                       ARRAY_N: a numerically indexed array (0 => value, ...),
+ *                       OBJECT: an object ( ->column = value ), respectively.
+ *
+ *                       With OBJECT_K: return an associative array of row objects keyed by the value
+ *                       of each row's first column's value.
+ */
```
Comment (szepeviktor) [inline tests/data/wpdb.php:27]:
```suggestion
```

This was just a helper for me to think.

**Comment by szepeviktor** [comment]
Thank you.
Will you include `wpdb::get_row`?

**Comment by herndlm** [comment]
that's only a slight improvement, but yeah I'll add it, give me a sec

**Comment by szepeviktor** [comment]
We do not have functional tests.
So 🙈 I close my eyes and ...

**Comment by herndlm** [comment]
It will be fine ™
BTW there's no magic way of releasing this before the next wp release, right?

**Comment by szepeviktor** [comment]
There is a way but in that case I will brake my promise to match WP.org versions.

**Comment by herndlm** [comment]
fyi, I switched to the `6.4.x-dev` tag in our app and can confirm that the issues I experienced there are solved now

**Comment by szepeviktor** [comment]
Thank you.

**Pull request merged**